code/__HELPERS/unsorted.dm

```diff
@@ -1095,36 +1083,54 @@ proc/get_mob_with_client_list()
 		else
 			return zone
 
+/*
+	get_turf(): Returns the turf that contains the atom.
+	Example: A fork inside a box inside a locker will return the turf the locker is standing on.
+	The weird for loop with an empty statement is apparently the fastest way possible to do this.
+*/
 /proc/get_turf(const/atom/O)
-	if (isnull(O) || isarea(O) || !istype(O))
-		return
+	ASSERT(istype(O) && !isarea(O))
 	var/atom/A
 	for(A=O, A && !isturf(A), A=A.loc);  // semicolon is for the empty statement
 	return A
 
-/proc/get(atom/loc, type)
-	while(loc)
-		if(istype(loc, type))
-			return loc
-		loc = loc.loc //<wwjnc> WE ARE THE KNIGHTS WHO SAY LOC
+/*
+	get_holder_at_turf_level(): Similar to get_turf(), will return the "highest up" holder of this atom, excluding the turf.
+	Example: A fork inside a box inside a locker will return the locker. Essentially, get_just_before_turf().
+*/
+/proc/get_holder_at_turf_level(const/atom/movable/O)
+	ASSERT(istype(O))
+	var/atom/A
+	for(A=O, A && !isturf(A.loc), A=A.loc);  // semicolon is for the empty statement
+	return A
+
+/*
+	get_holder_of_type(): Returns the FIRST holder of type specified. NOT the "highest up".
+	Example: Call find_holder_of_type(A, /mob) to find the first mob holder of A.
+*/
+/proc/get_holder_of_type(const/atom/movable/O, type)
+	ASSERT(istype(O))
+	var/atom/A = O
+	while(A && !isturf(A))
+		if(istype(A, type))
+			return A
+		A = A.loc
 	return null
 
-/proc/find_holder(atom/O) //aka get_just_before_turf
+/*
+	is_holder_of(): Returns 1 if A is a holder of B, meaning, A is B.loc or B.loc.loc or B.loc.loc.loc etc.
+	This is essentially the same as calling recursive_type_check(A) and checking if B is in the resulting list, but a lot faster and clearer to what you're doing.
+*/
+/proc/is_holder_of(const/atom/movable/A, const/atom/movable/B)
```

Comment:
Is there any reason why these type of procs are not an /atom/proc/? Seems like A.is_holder_of(B) would be clearer than is_holder_of(A,B), but all other procs stay away from that format